### DISCUSSION TITLE
Question about Regarding the Format Returned by Programming Language Snippet

### DISCUSSION THREAD

**Discussion by anthonyy41** [microsoft/teams-sdk#230] [Q&A] [answered]
Hello, I am Anthony. 

I followed the official documentation to implement a chat bot. (https://learn.microsoft.com/en-us/microsoftteams/platform/sbs-botbuilder-conversation-ai?tabs=vscode%2Cviscode) 
However, when I ask about programming language-related questions, the responses displayed on Teams contain HTML tags. 

![image](https://github.com/microsoft/teams-ai/assets/31025028/610d64f6-9009-4d15-b0c4-bbc7c023f6ac)

How can I adjust this so that these tags do not appear? Thank you~

**Comment by corinagum** [comment]
@anthonyy41 are you using the teamsd-ai library? We'll need more information in order to help you.

1. Do you want this response to be in plain text? At first glance, this seems like an HTML formatting error because `<code>` is not being used.
1. What feature of the teams-ai repo are you using, and what does the code that leads to this invoke look like?
2. How is the response generated? Via AI? OpenAI or AOAI? Does your skprompt have directions format the response in the way you are expecting? How does your code use the response from AI?
3. What does the full response look like (including the 'In this example' section?

**Comment by anthonyy41** [comment]
hi @corinagum I apologize for the delayed response as I have been busy these past few days.

And thank you for your response. Let me provide additional details about the testing process and information:
1. The code is based on the 04.ai.a.teamsChefBot project.
2. The part of the code that was modified is the defaultModel, which has been changed to gpt-3.5-turbo.
```js
// Create AI components
const planner = new OpenAIPlanner({
  apiKey: config.openAIKey,
  defaultModel: 'gpt-3.5-turbo',
  logRequests: true
});
```

3.  No other adjustments have been made to the rest of the code.

Additionally, here are the answers to your previous inquiries for your reference:
1. I would like the results to be returned in the following way (below is an example of different code and the displayed result).
```js
const chatCompletion = await openai.createChatCompletion({
        model: "gpt-3.5-turbo",
        messages: [{ role: "user", content: `${message.text}` }],
});

responseText = chatCompletion.data.choices[0].message;
await context.sendActivity(responseText.content);
```

![Pasted image 20230713091239](https://github.com/microsoft/teams-ai/assets/31025028/2256a950-2cc2-4193-a61f-4c4436fa6214)

2. I think the code causing this issue is the change of the model, but I'm not sure if this is the problem (text-davinci-003 => gpt-3.5-turbo).
3. I am using OpenAI's service and skprompt is base on example
4. Here is complete response.
![image](https://github.com/microsoft/teams-ai/assets/31025028/52a13826-06d6-450b-98de-0bf43552a400)

Please refer to the information above for reference. Thank you.

**Reply by corinagum** [reply]
Hi @anthonyy41, no problem about response time -- in fact, I'm juggling several tasks at once, so there will be a small delay to look into this. However, I have this discussion on my To-Do list, so please know that your response is on my radar :)

**Reply by anthonyy41** [reply]
Of course no problem, thank you for your assistance and look forward to the follow-up reply :)

**Comment by aacebo** [comment, marked as answer]
I ran through this test case and what chatgpt is responding with is markdown, so to get the desired result above I used the following code to convert the markdown to html and then convert that html to plain text.

![image](https://github.com/microsoft/teams-ai/assets/44811138/0832fb66-9137-42cf-bf3a-f3c94d170c81)

which yields:

![image](https://github.com/microsoft/teams-ai/assets/44811138/4c323459-8d48-4ee7-a006-df786958822d)

Does this answer your question @anthonyy41 ?

**Comment by anthonyy41** [comment]
hi @aacebo 
Thank you for your response and assistance. 
After testing, the presentation effect has indeed achieved what I wanted. 

In addition, this is another method I have adopted after referring to your suggestions (using activity.text.replace seems to maintain the style of code formatting). 

```js
server.post("/api/messages", async (req, res) => {
  await adapter.process(req, res as any, async (context) => {

    context.onSendActivities((ctx, activities, next) => {
      for (const activity of activities.filter(a => a.type === 'message')) {
        activity.text = activity.text.replace(/<br>/g, '\n'); // Replace <br> with \n
      }
      return next();
    });
    await app.run(context);
  });
});
```

![image](https://github.com/microsoft/teams-ai/assets/31025028/41338c2b-55e4-4edb-ae9f-09fa268480b5)

Finally, I am very grateful for the provision of this function library, which allows me to quickly get started with development needs on Teams.

**Reply by aacebo** [reply]
alright glad we could help, closing this issue.
